core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -813,56 +813,52 @@ class Partition(val topicPartition: TopicPartition,
    */
   private def tryCompleteDelayedRequests(): Unit = delayedOperations.checkAndCompleteAll()
 
-  private def shouldShrinkIsr(): Boolean = {
-    leaderLogIfLocal match {
+  def maybeShrinkIsr(): Unit = {
+    // Determine whether to shrink the ISR. Checking before attempting to do so removes the need to
+    // acquire a write lock on leaderIsrUpdateLock in the common case of no change.
+    val shouldShrinkIsr = leaderLogIfLocal match {
```

Comment:
Sorry if I wasn't super clear about this. I think we can keep shouldShrinkIsr as a help method. I mostly thought we didn't need this separate method here https://github.com/apache/kafka/pull/7921/commits/c46d0801d89c020b0e1fa831ff3c455a30c52c80#diff-5450baca03f57b9f2030f93a480e6969R863, and then we could make a comment about the read lock when we check it at the if statement here https://github.com/apache/kafka/pull/7921/commits/86138034706251efb40f8dcc2357494848cabbe7#diff-5450baca03f57b9f2030f93a480e6969R826. What do you think?

Reply:
Sure, that's fine. Updated.